streams/src/main/java/org/apache/kafka/streams/state/internals/WrappedStateStore.java

```diff
@@ -19,81 +19,66 @@
 import org.apache.kafka.streams.errors.InvalidStateStoreException;
 import org.apache.kafka.streams.processor.ProcessorContext;
 import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.TimestampedBytesStore;
 
 /**
  * A storage engine wrapper for utilities like logging, caching, and metering.
  */
-public interface WrappedStateStore extends StateStore {
-
-    /**
-     * Return the inner most storage engine
-     *
-     * @return wrapped inner storage engine
-     */
-    StateStore inner();
-
-    /**
-     * Return the state store this store directly wraps
-     * @return the state store this store directly wraps
-     */
-    StateStore wrappedStore();
-
-    abstract class AbstractStateStore implements WrappedStateStore {
-        final StateStore innerState;
-
-        protected AbstractStateStore(final StateStore inner) {
-            this.innerState = inner;
+public class WrappedStateStore<S extends StateStore> implements StateStore {
```

Comment:
There was no advantage to separating the interface and abstract class after adding the generic parameter, so I've collapsed them into one abstract class.

Previously, the interface was nice because some implementations could declare that they were wrapped without being forced to store duplicate references to the inner store.

Reply:
Good call!

Reply:
> collapsed them into one abstract class

Class is not declared as `abstract` -- is this intentionally or did it slip? IMHO, it should be `abstract` (should the name indicate this, too? -> `AbstractedWrappedStateStore`?)

Reply:
I can make it abstract; I don't think there's any advantage to just wrapping a state store, except for testing.

I don't think we need to change the name of the class, though...